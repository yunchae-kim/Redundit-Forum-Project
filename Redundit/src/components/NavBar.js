/* eslint-disable react/button-has-type */
import { useState, useEffect } from "react";
import { Button, Navbar, Alignment } from "@blueprintjs/core";
import { Link, useHistory } from "react-router-dom";
import httpClient from "../utility/auth.helper";
import logo from "../resources/logo.svg";
import userIcon from "../resources/userIcon.png";
import messageIcon from "../resources/messageIcon.png";
import "../style/components/NavBar.css";

const NavBar = () => {
  const history = useHistory();
  const [userId, setUserId] = useState(0);

  async function handleLogout() {
    httpClient.logOut();
    setUserId(0);
    history.push("/");
    history.go();
  }

  useEffect(() => {
    const userObj = httpClient.getCurrentUser();
    if (userObj) {
      const { iduser } = userObj;
      if (iduser) {
        setUserId(iduser);
      }
    }
  }, [userId]);

  return (
    <Navbar className="bp3-navbar navbar" fixedToTop>
      <Navbar.Group align={Alignment.LEFT}>
        <Link to="/">
          <img className="navbar-logo" src={logo} alt="" />
        </Link>
      </Navbar.Group>
      <Navbar.Group align={Alignment.RIGHT}>
        {userId
          ? <Button className="navbar-button" text="Log Out" intent="primary" small onClick={() => handleLogout()} />
          : (
            <div>
              <Link to="/login">
                <Button className="navbar-button" text="Log In" intent="primary" small />
              </Link>
              <Link to="/signup">
                <Button className="navbar-button" text="Sign Up" intent="primary" small />
              </Link>
            </div>
          )}
        {userId
          ? (
            <div>
              <Link to={{ pathname: `/user/${userId}`, state: { userId } }}><img className="navbar-icon" alt="" src={userIcon} /></Link>
              <Link to="/chat"><img className="navbar-icon navbar-icon-message" alt="" src={messageIcon} /></Link>
            </div>
          )
          : <></>}
      </Navbar.Group>
    </Navbar>
  );
};

export default NavBar;
