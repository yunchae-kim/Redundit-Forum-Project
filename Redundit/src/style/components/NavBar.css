.navbar.bp3-navbar.bp3-fixed-top{
  padding-bottom: 10px;
  padding-top: 0px;
  margin-top: 0px;
  width: 100%;
  overflow: hidden;
  position: fixed;
  height: 64px;
}

.bp3-button.bp3-intent-primary.navbar-button {
  border: none;
  width: 160px;
  height: 40px;
  border-radius: 20px;
  align-self: center;
  margin-right: 32px;
  justify-self: center;
  font-size: 20px;
  box-shadow: none;
  font-weight:600;
}

.navbar-logo {
  border-radius: 50%;
  width: 160px;
  height: 160px;
  align-self: center;
  margin-right: 32px;
  margin-left: 56px;
}

.navbar-icon {
  height: 30px;
  margin-right: 32px;
}

.navbar-icon-message {
  height: 30px;
  margin-right: 52px;
}
