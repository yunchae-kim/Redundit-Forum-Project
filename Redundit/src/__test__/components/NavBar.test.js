import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import NavBar from "../../components/NavBar";

test("test should NavBar match snapshot", async () => {
  const tree = renderer.create(
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
