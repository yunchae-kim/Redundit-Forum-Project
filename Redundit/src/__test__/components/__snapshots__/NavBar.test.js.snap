// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should NavBar match snapshot 1`] = `
<div
  className="bp3-navbar bp3-fixed-top bp3-navbar navbar"
>
  <div
    className="bp3-navbar-group bp3-align-left"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="navbar-logo"
        src="logo.svg"
      />
    </a>
  </div>
  <div
    className="bp3-navbar-group bp3-align-right"
  >
    <div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="bp3-button bp3-small bp3-intent-primary navbar-button"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="bp3-button-text"
          >
            Log In
          </span>
        </button>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="bp3-button bp3-small bp3-intent-primary navbar-button"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="bp3-button-text"
          >
            Sign Up
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;
